File: fns-hl7-pipeline/fn-mmg-validator/function/src/main/kotlin/mmgvalidator/HL7Message.kt
package com.example

import com.google.gson.annotations.SerializedName

import gov.nist.validation.report.Entry

data class HL7Message (

  @SerializedName("content"  ) var content  : String,
  @SerializedName("metadata" ) var metadata : HL7MessageMetadata,
  @SerializedName("structuralValidationReport" ) var structuralValidationReport : Map<String,List<Entry>>,
  // added in this fn
  @SerializedName("contentValidationReport" ) var contentValidationReport : Map<String,String>? = null,  
  
) // .HL7Message

File: fns-hl7-pipeline/fn-structural-validator/src/main/java/structruralvalidator/HL7Message.scala
package cdc.ede.hl7.structuralvalidator

import gov.nist.validation.report.Report

case class HL7MessageMetadata(filePath: String,
                                fileName: String, 
                                fileTimestamp: String, 
                                fileSize: Long, 
                                fileUUID: String, 
                                fileEventID: String, 
                                fileEventTimestamp: String, 
                                messageUUID: String,
                                messageIndex: Int)


case class HL7Message(content: String, metadata: HL7MessageMetadata)

                     
case class HL7MessageOut(content: String, metadata: HL7MessageMetadata, structuralValidationReport: Report)
